File: C Language/fibonacci_calc.c
/**
 * fibonacci_calc.c
 * @author Gabriel Rivera Per-ossenkopp
 * Student Number: 841-11-6930
 * September 24, 2014
 * Description: This program calculates the n-th term of the fibonacci sequence
 */

 #include <stdio.h>

 //prototype
 int iterative_fibo(int n);
 int recursive_fibo(int n);

 int main(void){
    int n;
    printf("Enter a non-negative integer: ");
    scanf("%d", &n);

    printf("\n");
    printf("Using iteration, the %d-th number in the Fibonacci sequence is %d\n", n, iterative_fibo(n));
    printf("Using recursion, the %d-th number in the Fibonacci sequence is %d\n", n, recursive_fibo(n));
    printf("Thanks for using this program.\n");

    return 0;
 }

/**
 * Returns the n-th number of the Fibonacci Sequence using iterations
 */
 int iterative_fibo(int n){
    int firstTerm = 0, secondTerm = 1, nextTerm = 0;
    while(n-- > 0){
        nextTerm = firstTerm + secondTerm;
        firstTerm = secondTerm;
        secondTerm = nextTerm;
    }

    return firstTerm;
 }

/**
 * Returns the n-th number of the Fibonacci Sequence using recursion
 */
 int recursive_fibo(int n){
    if(n == 0)
        return 0;
    if(n == 1)
        return 1;
    return recursive_fibo(n - 1) + recursive_fibo(n - 2);
 }

File: Fsharp Language/fibonacci_calc.fs
(*
 * fibonacci_calc.fs
 * @author Gabriel Rivera Per-ossenkopp
 * November 7th, 2014
 * Description: This program executes the fibonacci sequence
 *)

let fibo_tail num = 
    let rec helper fterm sterm num = 
        match num with
        | 0 -> fterm
        | _ -> helper sterm (fterm + sterm) (num - 1)
    helper 0 1 num

let rec fibo_rec num =
    match num with
    | 0 -> 0
    | 1 -> 1
    | _ -> fibo_rec (num - 1) + fibo_rec (num - 2)

let start num =
    printfn "The number is %d" num
    if num >= 0 then
        printfn "Using tail-recursion the %d-th fibonacci sequence is %d" num (fibo_tail num)
        printfn "Using regular-recursion the %d-th fibonacci sequence is %d" num (fibo_rec num)
    else
        printfn "Error! You must use a positive integer"
